src/Talismans.ts

```diff
@@ -456,20 +461,21 @@ export const buyTalismanLevels = (i: number, auto = false): boolean => {
             player.legendaryFragments -= priceMult * Math.max(0, Math.floor(1 + 1 / 192 * Math.pow(player.talismanLevels[i-1] - 150, 3)))
             player.mythicalFragments -= priceMult * Math.max(0, Math.floor(1 + 1 / 1280 * Math.pow(player.talismanLevels[i-1] - 150, 3)))
             player.talismanLevels[i-1] += 1;
-
+            hasPurchased = true;
         } else {
-            return false;
+            purchasedMax = false;
+            break;
         }
     }
 
-    if (!auto) {
+    if (!auto && hasPurchased) {
         showTalismanPrices(i);
         // When adding game state recalculations, update the talisman autobuyer in tack() as well
         updateTalismanInventory();
         calculateRuneLevels();
     }
 
-    return true;
+    return purchasedMax;
```

Comment:
Which would mean this would go to `hasPurchased` instead

Reply:
done